src/main/java/org/spongepowered/api/map/color/MapPalette.java

```diff
@@ -0,0 +1,131 @@
+    public static final MapColor LIGHT_BLUE = DummyObjectProvider.createFor(MapColor.class, "LIGHT_BLUE");
+    public static final MapColor YELLOW = DummyObjectProvider.createFor(MapColor.class, "YELLOW");
+    public static final MapColor LIME = DummyObjectProvider.createFor(MapColor.class, "LIME");
+    public static final MapColor PINK = DummyObjectProvider.createFor(MapColor.class, "PINK");
+    public static final MapColor GRAY = DummyObjectProvider.createFor(MapColor.class, "GRAY");
+    public static final MapColor SILVER = DummyObjectProvider.createFor(MapColor.class, "SILVER");
+    public static final MapColor CYAN = DummyObjectProvider.createFor(MapColor.class, "CYAN");
+    public static final MapColor PURPLE = DummyObjectProvider.createFor(MapColor.class, "PURPLE");
+    public static final MapColor BLUE = DummyObjectProvider.createFor(MapColor.class, "BLUE");
+    public static final MapColor BROWN = DummyObjectProvider.createFor(MapColor.class, "BROWN");
+    public static final MapColor GREEN = DummyObjectProvider.createFor(MapColor.class, "GREEN");
+    public static final MapColor RED = DummyObjectProvider.createFor(MapColor.class, "RED");
+    public static final MapColor BLACK = DummyObjectProvider.createFor(MapColor.class, "BLACK");
+    public static final MapColor GOLD = DummyObjectProvider.createFor(MapColor.class, "GOLD");
+    public static final MapColor DIAMOND = DummyObjectProvider.createFor(MapColor.class, "DIAMOND");
+    public static final MapColor LAPIS = DummyObjectProvider.createFor(MapColor.class, "LAPIS");
+    public static final MapColor EMERALD = DummyObjectProvider.createFor(MapColor.class, "EMERALD");
+    public static final MapColor OBSIDIAN = DummyObjectProvider.createFor(MapColor.class, "OBSIDIAN");
+    public static final MapColor NETHERRACK = DummyObjectProvider.createFor(MapColor.class, "NETHERRACK");
+
+    // SORTFIELDS:OFF
+
+    /**
+     * Returns all the possible {@link MapColor}s available, including the
+     * shaded colors.
+     *
+     * @return All the map colors including shades
+     */
+    public static ImmutableCollection<MapColor> getAll() {
+        return factory.getAll();
+    }
+
+    /**
+     * Returns all the possible named {@link MapColor}s available, this is
+     * basically just all the static entries in {@link MapPalette} but in an
+     * {@link ImmutableCollection<MapColor>}.
+     *
+     * @return All the named map colors
+     */
+    public static ImmutableCollection<MapColor> getAllNamed() {
+        return factory.getAllNamed();
+    }
+
+    /**
+     * Returns a {@link MapColor} from all the colors (including shades) based
+     * on the palette index. This allows easy lookup of colors from a raw map.
+     *
+     * @param colorIndex The index of the color in the map palette
+     * @return The object represented by that color
+     */
+    public static MapColor getByIndex(int colorIndex) {
```

Comment:
It would be nice to also be able to pass a `Color` here and find the closest matching `MapColor`. (Same could be said for `TextColor` and any other "colors" now that I think of it.)